src/components/Shared/ExpensesCostCenters.js

```diff
@@ -0,0 +1,175 @@
+            { from: 'recordId', to: 'ccId' },
+            { from: 'name', to: 'ccName' },
+            { from: 'reference', to: 'ccRef' }
+          ],
+          columnsInDropDown: [
+            { key: 'reference', value: 'Reference' },
+            { key: 'name', value: 'Name' }
+          ]
+        }
+      },
+      {
+        component: 'textfield',
+        label: labels.name,
+        name: 'ccName',
+        props: {
+          readOnly: true
+        }
+      },
+      {
+        component: 'textfield',
+        label: labels.amount,
+        name: 'amount',
+        props: {
+          readOnly: false
+        }
+      },
+  ]
+
+
+    const totalAmount = formik.values?.costCenters?.reduce((amount, row) => {
+      const amountValue = parseFloat(row.amount?.toString().replace(/,/g, '')) || 0
+
+      return amount + amountValue
+    }, 0)
+
+
+    const balance = row.amount - totalAmount
+
+    useEffect(() => {
+        formik.setFieldValue('balance', balance)
+    }, [balance])
+
+
+  return (
+    <FormShell
+      resourceId={ResourceIds.PaymentVoucherExpenses}
+      form={formik}
+      maxAccess={maxAccess}
+      isCleared={false}
+      isInfo={false}
+    >
+      <VertLayout>
+        <Fixed>
+          <Grid container spacing={4}>
+            <Grid item xs={12}>
+                <CustomTextField
+                    name='amount'
+                    label={labels.amount}
+                    value={row.amount}
+                    maxAccess={maxAccess}
+                    readOnly
+                    onChange={formik.handleChange}
+                    onClear={() => formik.setFieldValue('amount', '')}
+                    error={formik.touched.amount && Boolean(formik.errors.amount)}
+                />
+            </Grid>
+            <Grid item xs={12}>
+                <CustomTextField
+                    name='amountAssigned'
+                    label={labels.amountAssigned}
+                    value={totalAmount}
+                    maxAccess={maxAccess}
+                    readOnly
+                />
+            </Grid>
+            <Grid item xs={12}>
+                <CustomTextField
+                    name='balance'
+                    label={labels.balance}
+                    value={formik.values.balance}
+                    maxAccess={maxAccess}
+                    readOnly
+                    onChange={formik.handleChange}
+                    onClear={() => formik.setFieldValue('balance', '')}
+                    error={formik.touched.balance && Boolean(formik.errors.balance)}
+                />
+            </Grid>
+          </Grid>
+        </Fixed>
+        <Grow>
+          <DataGrid
+            onChange={value => {
+              formik.setFieldValue('costCenters', value)
+            }}
+            value={formik.values.costCenters}
+            error={formik.errors.costCenters}
+            columns={columns}
+            allowDelete={true}
+            allowAddNewLine={true}
```

Comment:
u set allowDelete and allowAddNewLine always as true, this should behave according to isCancel or isPosted